hazelcast/src/main/java/com/hazelcast/internal/util/phonehome/MapInfoCollector.java

```diff
@@ -55,52 +60,97 @@
+        return mapConfigs.stream().map(MapConfig::getWanReplicationRef)
+                .filter(Objects::nonNull).count();
     }
 
     private long countMapWithAtleastOneAttribute() {
-        return mapConfigs.stream()
-                .filter(mapConfig -> mapConfig != null && !(mapConfig.getAttributeConfigs().isEmpty())).count();
+        return mapConfigs.stream().map(MapConfig::getAttributeConfigs)
+                .filter(attributeConfigs -> !(attributeConfigs.isEmpty())).count();
     }
 
     private long countMapUsingEviction() {
-        return mapConfigs.stream().filter(mapConfig -> mapConfig != null
-                && mapConfig.getEvictionConfig().getEvictionPolicy() != EvictionPolicy.NONE).count();
+        return mapConfigs.stream().map(MapConfig::getEvictionConfig)
+                .filter(evictionConfig -> evictionConfig.getEvictionPolicy() != EvictionPolicy.NONE).count();
     }
 
     private long countMapWithNativeInMemoryFormat() {
-        return mapConfigs.stream()
-                .filter(mapConfig -> mapConfig != null && mapConfig.getInMemoryFormat() == InMemoryFormat.NATIVE).count();
+        return mapConfigs.stream().map(MapConfig::getInMemoryFormat)
+                .filter(inMemoryFormat -> inMemoryFormat == InMemoryFormat.NATIVE).count();
+    }
+
+    static Predicate<MapConfig> isMapStoreEnabled() {
+        return p -> p.getMapStoreConfig().isEnabled();
+    }
+
+    static Predicate<MapConfig> isMapStoreDisabled() {
+        return p -> !p.getMapStoreConfig().isEnabled();
+    }
+
+    private long mapPutLatency(Node node, Predicate<MapConfig> predicate) {
+        final long[] latencyInfo = {0, 0};
+        mapConfigs.stream()
+                .filter(predicate)
+                .map(mapConfig -> node.hazelcastInstance.getMap(mapConfig.getName()).getLocalMapStats())
+                .filter(mapStats -> mapStats.getPutOperationCount() != 0L)
+                .forEach(mapStats -> {
+                    latencyInfo[0] += mapStats.getTotalPutLatency();
+                    latencyInfo[1] += mapStats.getPutOperationCount();
+                });
+        if (latencyInfo[1] == 0) {
+            return -1;
+        }
+
+        return latencyInfo[0] / latencyInfo[1];
+    }
+
+    private long mapGetLatency(Node node, Predicate<MapConfig> predicate) {
+        final long[] latencyInfo = {0, 0};
+        mapConfigs.stream()
+                .filter(predicate)
+                .map(mapConfig -> node.hazelcastInstance.getMap(mapConfig.getName()).getLocalMapStats())
+                .filter(mapStats -> mapStats.getGetOperationCount() != 0L)
+                .forEach(mapStats -> {
+                    latencyInfo[0] += mapStats.getTotalGetLatency();
+                    latencyInfo[1] += mapStats.getGetOperationCount();
+                });
+        if (latencyInfo[1] == 0) {
+            return -1;
+        }
+        return latencyInfo[0] / latencyInfo[1];
```

Comment:
This two-element array is duplicated between the two methods. Can you extract this logic into a separate class?